### PR TITLE
FAI-6617 - Enhancing Octopus Deployments with Variable Sets

### PR DESCRIPTION
**Pull request by omreego**
## Description
This PR adds the Deployment Variable Set to the Streamed Deployment Object

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/octopus-source/resources/spec.json [hunk 1]
```diff
@@ -25,6 +25,14 @@
           "type": "string"
         }
       },
+      "variable_names": {
+        "type": "array",
+        "title": "Octopus Deployment Variable to Sync",
```
Comment (tovbinm) [inline sources/octopus-source/resources/spec.json:30]: `Variables` -> `Variables`

File: sources/octopus-source/resources/spec.json [hunk 2]
```diff
@@ -25,6 +25,14 @@
...
+        "description": "The Octopus Deployment Variables to sync.",
```
Comment (tovbinm) [inline sources/octopus-source/resources/spec.json:31]: Any additional details you can provide for the users?

Comment (tovbinm) [reply sources/octopus-source/resources/spec.json:31]: Does this get the variable values? Can these be sensitive? We should warn about this.

File: sources/octopus-source/src/octopusClient.ts [hunk 3]
```diff
@@ -145,6 +147,20 @@ export class OctopusClient {
...
+    spaceId: string,
+    variableSetId: string
+  ): Promise<OctopusDeploymentVariable[] | undefined> {
+    try {
+      const res: VariableSetResponse = await this.get<VariableSetResponse>(
+        `/${spaceId}/variables/${variableSetId}`
+      );
+      return res.Variables;
+    } catch (err: any) {
+      return undefined;
```
Comment (tovbinm) [inline sources/octopus-source/src/octopusClient.ts:160]: Should we at least log the error?

**Review by tovbinm — APPROVED** [review]

File: sources/octopus-source/src/octopus.ts [hunk 4]
```diff
@@ -151,15 +157,20 @@ export class Octopus {
...
+        const [project, environment, task, process, variables] =
+          await Promise.all([
+            this.client.getProject(deployment.ProjectId),
+            this.client.getEnvironment(deployment.EnvironmentId),
+            this.client.getTask(deployment.TaskId),
+            this.getDeploymentProcess(
+              deployment.ProjectId,
+              deployment.DeploymentProcessId
+            ),
+            this.getDeploymentVariables(
```
Comment (tovbinm) [inline sources/octopus-source/src/octopus.ts:169]: Let's avoid calling this API completely if no env deployment variable names were provided

**Review by tovbinm — APPROVED** [review]

File: sources/octopus-source/resources/schemas/deployments.json [hunk 5]
```diff
@@ -123,6 +123,43 @@
             }
           }
         }
+      },
+      "Variables": {
+        "type": "array",
+        "items": [
```
Comment (willmarks) [inline sources/octopus-source/resources/schemas/deployments.json:129]: I believe this should just be an object, not an array of objects.

**Pull request merged**